src/prefect/tasks/sftp/sftp.py

```
@@ -0,0 +1,240 @@
+
+    @defaults_from_attrs(
+        "host", "username", "password", "port_number", "remote_path", "local_path"
+    )
+    def run(
+        self,
+        host: str = None,
+        username: str = None,
+        password: str = None,
+        port_number: int = None,
+        remote_path: str = None,
+        local_path: str = None,
+    ):
+        """
+        Task for uploading files to an SFTP server.
+
+        Args:
+            - host (str): Name of the host to use.
+            - username (str): Username used to authenticate.
+            - password (str): Password used to authenticate.
+            - port_number (int): The port number to connect to the server.
+            - remote_path (str): The remote sftp file path.
+            - local_path (str): The local file path to upload from.
+
+        Raises:
+            - ValueError: Raised if a required parameter is not supplied.
+            - ClientError: Raised if exception occurs when connecting/uploading from the server.
+        """
+        if not host:
+            raise ValueError("A host name must be provided")
+        if not username:
+            raise ValueError("User name must be provided")
+        if not password:
+            raise ValueError("A password must be provided")
+        if not port_number:
+            raise ValueError("A port_number name must be provided")
+        if not remote_path:
+            raise ValueError("A remote_path must be provided")
+        if not local_path:
+            raise ValueError("A local_path must be provided")
+
+        # first init connection to SFTP server
+        self._create_connection()
+
+        # upload
+        try:
+            self._connection.put(
+                localpath=local_path,
+                remotepath=remote_path,
+                confirm=True,
+            )
+
+        except IOError as e:
+            if e.errno == errno.ENOENT:
+                raise
```

You can remove this except block and allow all exceptions to be raised for the user to see. We don't want silent failures.